src/lib/dom-api-shady.html

```diff
@@ -81,6 +82,11 @@
         } else {
           TreeApi.Composed.appendChild(container, node);
         }
+      // if a fragment provoked distribution, clean up it's actual dom.
+      } else if (isFragment) {
```

Comment:
No, this is special handling specifically for fragments. When fragments are added to dom, all of their children are added.